tests/test_auth_endpoint.py

```diff
@@ -0,0 +1,54 @@
+from fastapi.testclient import TestClient
+from datetime import timedelta
+from jose import jwt
+
+
+from ai_document_search_backend.application import app
+from ai_document_search_backend.auth import create_access_token
+from ai_document_search_backend.auth import get_hashed_password, check_password
+from ai_document_search_backend.config import Settings
+
+
+client = TestClient(app)
+
+
+def test_valid_authentication():
+    response = client.post("/auth/token", data={"username": "marius", "password": 123})
+    assert response.status_code == 200
+
+
+def test_invalid_password():
+    response = client.post("/auth/token", data={"username": "marius", "password": 1231})
+    assert response.status_code == 400
+    assert response.json() == {"detail": "Incorrect username or password"}
+
+
+def test_invalid_username():
+    response = client.post("/auth/token", data={"username": "maris", "password": 123})
+    assert response.status_code == 400
+    assert response.json() == {"detail": "Incorrect username or password"}
+
+
+def test_hashing():
+    password = "password"
+    hashed_password = get_hashed_password(password)
+    assert check_password(password, hashed_password)
+
+
+def test_token():
+    SECRETKEY = Settings().secretkey
```

Comment:
Can you use env var here, delete `ai_document_search_backend/config.py` and generate `hashed_password`s in `fake_users_db` instead of hard coding them? That way we can easily change the secretkey.

Reply:
The secretkey is used for encoding the JWT token. I think the purpose is that by encoding a secret key into the token, we can prove that the token was issued by us. Hashed passwords are not dependent on the secretkey, so changing it should not have any effect on the passwords.

Reply:
Ah, my bad. I get it now.
But the point remains: Right now, the `secretkey: str = "hi"`, which we agree is not the strongest key. I assume you want to change it to read from the env var and the value should be what you sent me over Slack, right? So I'm saying you can do it already.
And also that you probably don't need the `Settings` class, as all it contains is this env var and another unused var. And we already have the `Container` class which reads the config.

Reply:
The secretkey is in the .env file, "hi" is a default that prevents the backend from crashing something goes wrong with the actual key, so this should work by adding the .env file to your environment. However, I used it only for debugging, so I agree that the = "hi" can be removed. As for the Settings class, I'll push a new commit where I move it to Container

Reply:
I've tried it out, putting the code in the `Container` class does not work, at least not without modifying the code more. I suggest using the `.env` file the way the [fastapi documentation](https://fastapi.tiangolo.com/advanced/settings/?h=.env#reading-a-env-file) does, which includes `Settings`. The "other unused variable" is what reads `.env`, and makes sure the actual secretkey is used.

Reply:
> "hi" is a default that prevents the backend from crashing something goes wrong with the actual key

It's definitely better to let it crash than having such an insecure key by mistake.

Reply:
Yep, the most recent commit removed the default key